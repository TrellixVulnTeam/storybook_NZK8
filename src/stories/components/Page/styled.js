// @flow

import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
`;

export const InnerWrapper = styled.div`
  width: 100%;
  overflow-y: ${p => (p.disableScroll ? 'auto' : 'auto')};
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: ${p => (p.centered ? 'center' : 'flex-start')};

  padding: ${p => (p.withMargin ? '10px' : 0)};

  @media (min-width: 1200px) {
    padding: ${p => (p.withMargin ? '30px' : 0)};
    padding-top: ${p => (p.withMargin ? '16px' : 0)};
  }
`;
