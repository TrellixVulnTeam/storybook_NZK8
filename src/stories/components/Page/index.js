// @flow

import React from 'react';
import { Wrapper, InnerWrapper } from './styled';

type Props = {
  withMargin?: boolean,
  disableScroll?: boolean,
  centered?: boolean,
  children: React$Node,
};

export default (props: Props) => (
  <Wrapper>
    <InnerWrapper
      className="_test-inner-wrapper"
      withMargin={props.withMargin}
      centered={props.centered}
      disableScroll={props.disableScroll}
    >
      {props.children}
    </InnerWrapper>
  </Wrapper>
);
