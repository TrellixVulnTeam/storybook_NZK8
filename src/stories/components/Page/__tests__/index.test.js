import React from 'react';
import { shallow } from 'enzyme';
import Page from '../index';
import 'jest-styled-components';

describe('<Page />', () => {
  it('snapshot', () => {
    const c = shallow(<Page />);

    expect(c).toMatchSnapshot();
  });

  it('should render props.children', () => {
    const c = shallow(
      <Page>
        <div id="foo" />
      </Page>,
    );

    expect(c.contains(<div id="foo" />)).toStrictEqual(true);
  });

  it('should handle props.withMargin', () => {
    const c = shallow(
      <Page>
        <div />
      </Page>,
    );

    expect(c.find('._test-inner-wrapper')).toHaveStyleRule('padding', '0');

    c.setProps({
      withMargin: true,
    });

    expect(c.find('._test-inner-wrapper')).toHaveStyleRule('padding', '10px');
  });

  it('should handle props.centered', () => {
    const c = shallow(
      <Page>
        <div />
      </Page>,
    );

    expect(c.find('._test-inner-wrapper')).toHaveStyleRule('align-items', 'flex-start');

    c.setProps({
      centered: true,
    });

    expect(c.find('._test-inner-wrapper')).toHaveStyleRule('align-items', 'center');
  });
});
