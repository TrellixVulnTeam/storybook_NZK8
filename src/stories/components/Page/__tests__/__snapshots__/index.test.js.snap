// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> snapshot 1`] = `
<styled__Wrapper>
  <styled__InnerWrapper
    className="_test-inner-wrapper"
  />
</styled__Wrapper>
`;
